blobfuse/src/BlobfuseGcCache.cpp

```diff
@@ -162,8 +171,10 @@ void gc_cache::run_gc_cache()
 
             struct stat buf;
             stat(mntPath, &buf);
-            if (((now - buf.st_mtime) > file_cache_timeout_in_seconds) ||
-                disk_threshold_reached)
+            if (file_cache_timeout_in_seconds == 0 ||
+                disk_threshold_reached ||
+                file.force ||
+                (((now - buf.st_mtime) > file_cache_timeout_in_seconds) && ((now - buf.st_ctime) > file_cache_timeout_in_seconds)))
```

Comment:
ctime is not needed, if it is greater than modified time then it is greater than create time.

Reply:
ctime is change time and we can not rely on mod-time alone because in case of adls mod time reflects the last modified time of the file as per container and it can be long back then current time. Change time is updated everytime we download the file as we set its mod. Hence relying on ctime is more accurtae. In azs_open where we decide to redownload the file same logic is applied by checking both the times.